integration-tests/test/z-batch-serialization.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { SequencerBatch, BatchType } from '@eth-optimism/core-utils'
+
+import { expect } from './shared/setup'
+import { OptimismEnv } from './shared/env'
+import { envConfig } from './shared/utils'
+
+describe('Batch Serialization', () => {
+  let env: OptimismEnv
+  // Allow for each type to be tested. The env var here must be
+  // the same value that is passed to the batch submitter
+  const batchType = envConfig.BATCH_SUBMITTER_SEQUENCER_BATCH_TYPE.toUpperCase()
+  before(async () => {
+    env = await OptimismEnv.new()
+  })
+
+  it('should fetch batches', async () => {
+    const tip = await env.l1Provider.getBlockNumber()
+    const logs = await env.ctc.queryFilter(
+      env.ctc.filters.TransactionBatchAppended(),
+      0,
+      tip
+    )
+    // collect all of the batches
+    const batches = []
+    for (const log of logs) {
+      const tx = await env.l1Provider.getTransaction(log.transactionHash)
+      batches.push(tx.data)
+    }
+
+    expect(batches.length).to.be.gt(0, 'Submit some batches first')
+
+    let latest = 0
+    // decode all of the batches
+    for (const batch of batches) {
+      // Typings don't work?
+      const decoded = (SequencerBatch as any).fromHex(batch)
```

Comment:
Its not clear to me as to why the typings don't work here, I think it has to do with configuring ts to look for typings, the types work within the package but not outside of it

Reply:
We need to publish a types package for https://github.com/bcoin-org/bufio